plume-models/src/lib.rs

```diff
@@ -63,22 +70,41 @@ macro_rules! get {
 
 macro_rules! insert {
     ($table:ident, $from:ident) => {
-        pub fn insert(conn: &PgConnection, new: $from) -> Self {
+        last!($table);
+
+        pub fn insert(conn: &crate::Connection, new: $from) -> Self {
             diesel::insert_into($table::table)
                 .values(new)
-                .get_result(conn)
-                .expect("Error saving new $table")
+                .execute(conn)
+                .expect("Error saving new $table");
+            Self::last(conn)
         }
     };
 }
 
 macro_rules! update {
     ($table:ident) => {
-        pub fn update(&self, conn: &PgConnection) -> Self {
+        pub fn update(&self, conn: &crate::Connection) -> Self {
             diesel::update(self)
                 .set(self)
-                .get_result(conn)
-                .expect(concat!("Error updating ", stringify!($table)))
+                .execute(conn)
+                .expect(concat!("Error updating ", stringify!($table)));
+            Self::get(conn, self.id)
+                .expect(concat!(stringify!($table), " we just updated doesn't exist anymore???"))
+        }
+    };
+}
+
+macro_rules! last {
```

Comment:
how come none of these macros have docs?

Reply:
I'll add them! :grin: